### PR TITLE
Secretary Pages & More

### PR DESCRIPTION
**Pull request by aligmic**
(Updated with Agamemnon changes)
1. Updated ViewCourseDetails.jsp (New view and able to display course teachers).
2. Secretary html and jsp pages addition.
3. Updated PersonDao, CourseDao, DAOFactory, Course and Person java files.

### CODE REVIEW

File: AcademicManagerV1/src/com/academic/db/dao/PersonDAO.java [hunk 1]
```diff
@@ -8,22 +8,36 @@
...
+		countStatement = dbConnection.prepareStatement("SELECT COUNT(*) FROM person;",
 				ResultSet.TYPE_SCROLL_INSENSITIVE, ResultSet.CONCUR_READ_ONLY);
+
+		addStatement = dbConnection.prepareStatement(
+				"INSERT INTO person (personId,name,surname,type,dateOfBirth,email,phoneNumber,address,taxNumber,bankAccount,sex) VALUES (null, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?);");
+
+		updateStatement = dbConnection.prepareStatement(
+				"UPDATE person SET name=?,surname=?,dateOfBirth=?,email=?,phoneNumber=?,address=?,taxnumber=?,bankAccount=?,sex=? WHERE personId= ?;");
+		
+		deleteStatement = dbConnection.prepareStatement("DELETE * FROM person WHERE personId=?");
```
Comment (gmetal) [inline AcademicManagerV1/src/com/academic/db/dao/PersonDAO.java:40]: Convert to soft delete (isDeleted=1). You will also need to make changes in the Database (?) and the other Select SQL statements

File: AcademicManagerV1/src/com/academic/db/dao/CourseDAO.java [hunk 2]
```diff
@@ -30,14 +32,18 @@ public CourseDAO(Connection conn) throws SQLException {
...
 		addStatement = dbConnection.prepareStatement(
 				"INSERT INTO course (courseId, title, cost, description, startingDate, endingDate, isActive, isDeleted) VALUES ('?','?','?','?','?','?','?','?')");
 
 		updateStatement = dbConnection.prepareStatement(
-				"UPDATE course SET title= ? , cost= ?, description= ?, startingDate= ?, endingDate= ? , isActive= ? WHERE courseId=?");
-
-		deleteStatement = dbConnection.prepareStatement("UPDATE course SET isDeleted=1 WHERE courseId=?");
+				"UPDATE course SET title=?,cost=?,description=?,startingDate=?,endingDate=?,isActive=? WHERE courseId= ?;");
+		
+		deleteStatement = dbConnection.prepareStatement("UPDATE course SET ,isDeleted=? WHERE courseId=?");
```
Comment (gmetal) [inline AcademicManagerV1/src/com/academic/db/dao/CourseDAO.java:46]: Have a look at the latest code in the repository. This has been modified. This statement was wrong.

File: AcademicManagerV1/WebContent/html/CourseForm.jsp [hunk 3]
```diff
@@ -0,0 +1,140 @@
+<!DOCTYPE html>
```
Comment (gmetal) [inline AcademicManagerV1/WebContent/html/CourseForm.jsp:1]: Remove this file for now. It has been assigned to another group.

File: AcademicManagerV1/WebContent/html/ViewAllStudents.jsp [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+	integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u"
+	crossorigin="anonymous">
+</head>
+<body>
+	<header>
+		<a href="./SecretaryMain.html">
+			<button class="btn btn-color" role="button">Back</button>
+		</a>
+		<a href="#">
+		<button class="btn btn-success pull-right">Add New Student</button>
```
Comment (gmetal) [inline AcademicManagerV1/WebContent/html/ViewAllStudents.jsp:23]: Will this actually work?

File: AcademicManagerV1/src/com/academic/db/dao/CourseDAO.java (hunk 2 above)
Comment (aligmic) [reply AcademicManagerV1/src/com/academic/db/dao/CourseDAO.java:46]: it seems i missed it somehow.

File: AcademicManagerV1/WebContent/html/CourseForm.jsp (hunk 3 above)
Comment (aligmic) [reply AcademicManagerV1/WebContent/html/CourseForm.jsp:1]: ok i will remove it.

File: AcademicManagerV1/WebContent/html/ViewAllStudents.jsp (hunk 4 above)
Comment (aligmic) [reply AcademicManagerV1/WebContent/html/ViewAllStudents.jsp:23]: Add new student button does not work.. yet..

File: AcademicManagerV1/src/com/academic/db/dao/PersonDAO.java (hunk 1 above)
Comment (aligmic) [reply AcademicManagerV1/src/com/academic/db/dao/PersonDAO.java:40]: So you want us to make all the necessary changes to the database. Ok!

**Pull request closed without merging**